### PR TITLE
[MODULAR] Dorms NIFSoft

### PR DESCRIPTION
**Pull request by softcerv**
## About The Pull Request
Adds in the Grimoire Purpura NIFSoft, a subtype of the Grimoire Caeruleam that summons objects for adult activities. 

This also fixes the error that I made with the original prop summoner code that made the `holographic_filter` variable effectively useless.

## How This Contributes To The Skyrat Roleplay Experience
It is always good to give adults more convenient ways to participate in activities that adults would like to participate in.

Aside from that, this NIFSoft should give people, that participate in adult activities, a way to participate in said activities when maps don't include the proper vendor. 

## Proof of Testing

<details>
<summary>Screenshots/Videos</summary>
  
image

image

image

image

</details>

## Changelog

:cl:
add: Adds in the Grimoire Purpura, a NIFSoft that lets the user summon items for adult activities. 
/:cl:

### CODE REVIEW

**Comment by W4rd3nn** [comment]
so mad u didnt call it the grimoire erotica

File: modular_skyrat/modules/modular_implants/code/nifsofts/dorms.dm [hunk 1]
```diff
@@ -0,0 +1,61 @@
+/obj/item/disk/nifsoft_uploader/dorms
+	name = "Grimoire Purpura"
+	loaded_nifsoft = /datum/nifsoft/summoner/dorms
+
+/datum/nifsoft/summoner/dorms
+	name = "Grimoire Purpura"
+	program_desc = "Grimoire Purapura, a fork of the Grimoire Caeruleam code, allows users to conveniently access an extensive database of various adult toys. "
```
Comment (tf-4) [inline modular_skyrat/modules/modular_implants/code/nifsofts/dorms.dm]:
```suggestion
	program_desc = "Grimoire Purpura, a fork of the Grimoire Caeruleam code, allows users to conveniently access an extensive database of various adult toys. "
```

File: modular_skyrat/modules/modular_implants/code/nifsofts/dorms.dm [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+	holographic_filter = FALSE //No RGB toys
+	name_tag = "purapura "
```
Comment (tf-4) [inline modular_skyrat/modules/modular_implants/code/nifsofts/dorms.dm]:
```suggestion
	name_tag = "purpura "
```

**Review by Zonespace27 — APPROVED** [review]

**Comment by Zonespace27** [comment]
NIF ownership suddenly spikes 300%

**Pull request merged**